docs/core/preview3/tutorials/using-on-mac-vs-full-solution.md

```diff
@@ -0,0 +1,109 @@
+keywords: .NET, .NET Core, macOS, Mac
+author: bleroy
+manager: wpickett
+ms.date: 11/16/2016
+ms.topic: article
+ms.prod: .net-core
+ms.technology: .net-core-technologies
+ms.devlang: dotnet
+ms.assetid: d743134a-08a3-4ff6-aab7-49f71f0568c3
+---
+
+# Building a complete .NET Core solution on macOS, using Visual Studio for Mac
+
+by [Bertrand Le Roy](https://github.com/bleroy)
+
+Visual Studio for Mac provides a full-featured development environment for developing .NET Core applications. The procedures in this document describe the steps necessary to build a typical .NET Core solution that includes reusable libraries, testing, and using third-party libraries.
+
+> Visual Studio for Mac is preview software
+
+## Prerequisites
+
+Follow the instructions on [our prerequisites page](../macos-prerequisites.md) to update your environment.
+
+# A solution using only .NET Core projects
+
+## Writing the library
+
+1. In Visual Studio, choose **New Project**. In the new project dialog, choose **Library** under **.NET Core**, then select **Class Library (.NET Core)**. Click **Next**.
+
+   ![Creating a new library in Visual Studio for Mac](./media/vsmacfull01.png)
+
+2. Name the project "Library" and the solution "Golden". Leave **Create a project directory within the solution directory** checked. Click **Create**.
+
+   ![Naming the library project](./media/vsmacfull02.png)
+   
+3. By default, the current project template sets the .NET Standard version to 1.4. We'll bump it up to 1.6. Right-click the project and choose **Tools / Edit File**. Change `<TargetFramework>netstandard1.4</TargetFramework>` to `<TargetFramework>netstandard1.6</TargetFramework>`. Save.
+
+4. In the solution explorer, open the context menu for the **Dependencies** node and choose **Add Packages...**.
+
+5. Choose "nuget.org" as the Package source, and check **Json.NET**. Click **Add Package**. The package should now appear under **Dependencies/NuGet** and be automatically restored. [Json.NET](http://www.newtonsoft.com/json) is the recommended library and what our code will use to perform [JSON](http://www.json.org/) serialization and deserialization.
+
+   ![Adding the Json.NET dependency to the library project](./media/vsmacfull03.png)
+
+6. Rename the `MyClass.cs` file to `Thing.cs`. Also rename the class and constructor `Thing`. Add a method: `public int Get(int number) => Newtonsoft.Json.JsonConvert.DeserializeObject<int>($"{number}");`
+
+7. On the **Build** menu, choose **Build All**.
+
+   The solution should build without error.
+
+### Writing the test project
+
+1. In the solution explorer, open the context menu for the Golden solution node and choose **Add**, **Add New Project...**. In the new project dialog, choose **.NET Core / App** (and not **Library**), and then **Console Application (.NET Core)**. Click **Next**. Name the new project "TestLibrary" and click **Create**.
+
+   ![Adding the test project](./media/vsmacfull05.png)
```

Comment:
We need to start adding better alt text (in general). Here's a good topic on this: https://worldready.cloudapp.net/StyleGuide/Read?id=2818&topicid=37020

So here the alt text should read something like: 
"Screenshot of the template selection dialog showing Console Application (.NET Core) option selected"